adam-core/src/main/scala/org/bdgenomics/adam/rdd/sequence/ReadRDD.scala

```diff
@@ -0,0 +1,198 @@
+
+    // get sequence dictionary
+    val sd = new SequenceDictionary(rdd.flatMap(read => {
+      if (read.getName != null && read.getLength != null) {
+        Some(SequenceRecord(read.getName, read.getLength))
+      } else {
+        None
+      }
+    }).distinct
+      .collect
+      .toVector)
+
+    ReadRDD(rdd, sd)
+  }
+}
+
+/**
+ * A GenomicRDD that wraps Read data.
+ *
+ * @param rdd An RDD of genomic Reads.
+ * @param sequences The reference genome this data is aligned to.
+ */
+case class ReadRDD(rdd: RDD[Read],
+                   sequences: SequenceDictionary) extends AvroGenomicRDD[Read, ReadRDD] with Logging {
+
+  protected def buildTree(rdd: RDD[(ReferenceRegion, Read)])(
+    implicit tTag: ClassTag[Read]): IntervalArray[ReferenceRegion, Read] = {
+    IntervalArray(rdd, ReadArray.apply(_, _))
+  }
+
+  /**
+   * Save reads as Parquet or FASTQ.
+   *
+   * If filename ends in .fq or .fastq, saves as FASTQ. If not, saves reads
+   * to Parquet.
+   *
+   * @param filePath Path to save files to.
+   * @param asSingleFile If true, saves output as a single file.
+   */
+  def save(filePath: java.lang.String, asSingleFile: java.lang.Boolean) {
+    if (filePath.endsWith(".fq") || filePath.endsWith(".fastq")) {
+      saveAsFastq(filePath, asSingleFile = asSingleFile)
+    } else {
+      if (asSingleFile) {
+        log.warn("asSingleFile = true ignored when saving as Parquet.")
+      }
+      saveAsParquet(new JavaSaveArgs(filePath))
+    }
+  }
+
+  /**
+   * Save reads in FASTQ format.
+   *
+   * @param filePath Path to save files to.
+   * @param asSingleFile If true, saves output as a single file.
+   */
+  def saveAsFastq(filePath: String, asSingleFile: Boolean = false) {
+
+    def toFastq(read: Read): String = {
+      val sb = new StringBuilder()
+      sb.append("@")
+      sb.append(read.getName)
+      Option(read.getDescription).foreach(n => sb.append(" ").append(n))
+      sb.append("\n")
+      sb.append(read.getSequence)
+      sb.append("\n+\n")
+      sb.append(read.getQualityScores)
+      sb.append("\n")
+      sb.toString
+    }
+
+    writeTextRdd(rdd.map(toFastq), filePath, asSingleFile)
+  }
+
+  /**
+   * @param newRdd The RDD to replace the underlying RDD with.
+   * @return Returns a new ReadRDD with the underlying RDD replaced.
+   */
+  protected def replaceRdd(newRdd: RDD[Read]): ReadRDD = {
+    copy(rdd = newRdd)
+  }
+
+  /**
+   * @param read Read to extract a region from.
+   * @return Returns a reference region that covers the entirety of the read.
+   */
+  protected def getReferenceRegions(read: Read): Seq[ReferenceRegion] = {
+    Seq(ReferenceRegion.all(read.getName))
+  }
+
+  /**
+   * Writes an RDD to disk as text and optionally merges.
+   *
+   * @param rdd RDD to save.
+   * @param outputPath Output path to save text files to.
+   * @param asSingleFile If true, combines all partition shards.
+   */
+  private def writeTextRdd[T](rdd: RDD[T],
```

Comment:
As an aside, this seems like it is used in a lot of places. I think we have this in `FragmentRDD` too? Perhaps we can move it into `GenomicRDD` as a protected method?